### DISCUSSION TITLE
Account/Case/Contact/Lead - Flow vs apex trigger?

### DISCUSSION THREAD

**Discussion by baobao917** [jamessimone/apex-rollup#512] [Q&A] [answered]
I saw this blurb buried in the special considerations:

![image](https://github.com/jamessimone/apex-rollup/assets/104730171/c0b90327-26bf-4ccb-a524-1808ff3865b3)

- Does this mean for those objects, even if you have setup flow triggers (After Create/Update) & (Before Delete), you still need an apex trigger for After Delete?  

- If above is yes, I assume then best practice would be to consolidate all (after insert, after update, before delete, and after undelete) of that into an apex trigger instead of having multiple flows and triggers?  

- If above is yes, suggestion that the Flow/Process Builder Invocable section highlight this in one or all of the below sections
![image](https://github.com/jamessimone/apex-rollup/assets/104730171/82def850-3b72-4c8f-a1ef-22bf02bf365c)

**Comment by jamessimone** [comment]
@baobao917 I've updated the documentation for this as well - thank you for your feedback! That will display when 1.5.88 is released.

**Comment by baobao917** [comment]
@jamessimone Thanks.  I did have the following question:

- I assume then best practice would be to consolidate all (after insert, after update, before delete, and after undelete) of that into an apex trigger instead of having multiple flows and triggers? (i.e. a Create/Update flow, a Before Delete flow, and an After Delete apex trigger)

I did see where you added that info (and pasting screenshot here for others to see).
![image](https://github.com/jamessimone/apex-rollup/assets/104730171/72964b3b-9e17-4604-90ad-cbfa564c65d1)

**Comment by jamessimone** [comment, marked as answer]
Yes - for Account / Case / Contact / Lead-based rollups where any of those objects is configured as the parent _and_ merges occur in your org, I would recommend consolidating everything into the Apex one-liner within a trigger with all of the appropriate contexts (after insert, after update, before delete, after delete, and after undelete if ... you do things like undelete!)

**Reply by baobao917** [reply]
That's what I thought.  Thanks for the confirmation!
